### PR TITLE
Health profile 4448

### PR DESCRIPTION
**Pull request by waicheng**
Added API documentation for health profile endpoints:
```
1. /health_profile
2. /health_profile_question
3. /health_profile_answer
4. /health_question_definition
```

### CODE REVIEW

File: spec/swagger.yaml [hunk 1]
```diff
@@ -4783,5 +5028,548 @@ definitions:
...
+        type: string
+      attributes:
+        type: object
+        properties:
+          stats:
+            type: array
+            items:
+              description: 'The category, answered and total counts for questions in the health profile'
+              properties:
+                category:
```
Comment (sgilroy) [inline spec/swagger.yaml]: Add enum for the possible values

File: spec/swagger.yaml [hunk 2]
```diff
@@ -4783,5 +5028,548 @@ definitions:
...
+        $ref: '#/definitions/CollectionResponseLinks'
+      data:
+        type: array
+        items:
+          $ref: '#/definitions/HealthProfileResource'
+      included:
+        description: |
+          Related resources which are included in the response based on the `include` param.
+          Attributes of each resource will vary depending on the type.
+          See [action](#operation/fetchHealthProfileQuestion), [bundle](#operation/fetchHealthProfileAnswer)
```
Comment (sgilroy) [inline spec/swagger.yaml]: Fix as per above

File: spec/swagger.yaml [hunk 3]
```diff
@@ -4783,5 +5028,548 @@ definitions:
...
+    required:
+      - type
+      - id
+    properties:
+      type:
+        type: string
+      id:
+        type: string
+      attributes:
+        type: object
```
Comment (sgilroy) [inline spec/swagger.yaml]: The absence of attributes probably warrants some explanation. Or perhaps attributes should be omitted? The example does not include `data.attributes`.

File: spec/swagger.yaml [hunk 4]
```diff
@@ -1537,6 +1549,239 @@ paths:
...
               $ref: '#/definitions/FetchErrorResponse'
 
+  /health_profile:
+    get:
+      tags:
+        - health profile
+      summary: List health profiles
+      description: Get a list of health profiles
+      operationId: fetchHealthProfiles
+      parameters:
```
[the remark this answers is not in the document]
Comment (sgilroy) [reply spec/swagger.yaml]: Resolved.

File: spec/swagger.yaml (hunk 3 above)
Comment (sgilroy) [reply spec/swagger.yaml]: Resolved.

File: spec/swagger.yaml [hunk 5]
```diff
@@ -4783,5 +5028,548 @@ definitions:
...
+    properties:
+      meta:
+        $ref: '#/definitions/FetchMetaResponse'
+      data:
+        $ref: '#/definitions/HealthProfileResource'
+      included:
+        description: |
+          Related resources which are included in the response based on the `include` param.
+          Attributes of each resource will vary depending on the type.
+          See [action](#operation/fetchHealthProfileQuestion), [bundle](#operation/fetchHealthProfileAnswer)
```
[the remark this answers is not in the document]
Comment (waicheng) [reply spec/swagger.yaml]: Updated to  See [patient](#operation/fetchPatient)`

**Comment by waicheng** [comment]
The `health_question_definition` resource does not have any relationship and does not allow `include` as parameter, therefore `fetchHealthQuestionDefinitions` and `fetchHealthQuestionDefinition` would not have an `included` element.

**Comment by waicheng** [comment]
@sgilroy Will you be merging this to master?

**Comment by sgilroy** [comment]
@waicheng You should merge this PR.

**Pull request merged**